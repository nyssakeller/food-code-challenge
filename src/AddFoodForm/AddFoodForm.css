input, button {
  font-size: 18px;
  margin-bottom: 25px;
}

input {
  height: 30px;
  margin-right: 7px;
  padding-left: 7px;
  width: 270px;
}

button {
  background-color: red;
  height: 36px;
  width: 80px;
}