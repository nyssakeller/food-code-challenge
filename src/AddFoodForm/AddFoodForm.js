import React, { Component } from 'react';

class AddFoodForm extends Component {
  constructor() {
    super();
    this.state = {
      inputValue: ''
    }
  }

  updateInputValue = e => {
    const { value } = e.target;
    this.setState({ inputValue: value });
  }

  render() {
    return(
      <div>
        <form>
          <input type='text'
                 value={ this.state.inputValue }
                 placeholder='Enter food here'
                 onChange={ (e) => this.updateInputValue(e) } />
          <button>Submit</button>
        </form>
      </div>
    );
  }
}

export default AddFoodForm;