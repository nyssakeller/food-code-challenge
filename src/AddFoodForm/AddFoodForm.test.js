import React from 'react';
import { shallow, mount } from 'enzyme';
import AddFoodForm from './AddFoodForm';

describe('AddFoodForm', () => {

  it('should match snapshot', () => {
    const wrapper = shallow( <AddFoodForm /> );
    expect(wrapper).toMatchSnapshot();
  });

  it('has an inital state of inputValue set to an empty string', () => {
    const wrapper = shallow( <AddFoodForm /> );
    expect(wrapper.state().inputValue).toEqual('');
  });
});