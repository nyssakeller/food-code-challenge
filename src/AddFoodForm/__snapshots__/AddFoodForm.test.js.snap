// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddFoodForm should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <AddFoodForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
      >
        <input
          onChange={[Function]}
          placeholder="Enter food here"
          type="text"
          value=""
        />
        <button>
          Submit
        </button>
      </form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            onChange={[Function]}
            placeholder="Enter food here"
            type="text"
            value=""
          />,
          <button>
            Submit
          </button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "placeholder": "Enter food here",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Submit",
          },
          "ref": null,
          "rendered": "Submit",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
        >
          <input
            onChange={[Function]}
            placeholder="Enter food here"
            type="text"
            value=""
          />
          <button>
            Submit
          </button>
        </form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              onChange={[Function]}
              placeholder="Enter food here"
              type="text"
              value=""
            />,
            <button>
              Submit
            </button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "onChange": [Function],
              "placeholder": "Enter food here",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Submit",
            },
            "ref": null,
            "rendered": "Submit",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
